src/runtime/containerd-shim-v2/service.go

```diff
@@ -710,6 +710,14 @@ func (s *service) Kill(ctx context.Context, r *taskAPI.KillRequest) (_ *ptypes.E
 			}).Debug("Id of exec process to be signalled is empty")
 			return empty, errors.New("The exec process does not exist")
 		}
+		if (signum == syscall.SIGKILL || signum == syscall.SIGTERM) && execs.status == task.StatusStopped {
```

Comment:
Can we merge the container's  check with this exec's check in one place, thus the code would be look neat.

Reply:
Sure, will merge them. Good point!